scripts/deploy.ts

```diff
@@ -0,0 +1,40 @@
+import * as dotenv from "dotenv";
+import { ethers, upgrades } from "hardhat";
+// eslint-disable-next-line node/no-missing-import
+import { CliptoExchange } from "../typechain";
+
+dotenv.config();
+
+/*
+ * this script deploys the exchange and token contracts
+ * along with their beacon and proxy.
+ */
+async function main() {
+  const feeDestination = "0xaDb10b8112Fac755e8ab1DfFaab116523844DD18";
+
+  const cliptoToken = await ethers.getContractFactory("CliptoToken");
+  const cliptoTokenBeacon = await upgrades.deployBeacon(cliptoToken);
+  await cliptoTokenBeacon.deployed();
+  const cliptoTokenAddress = cliptoTokenBeacon.address;
+
+  const cliptoExchange = await ethers.getContractFactory("CliptoExchange");
+  const cliptoExchangeProxy = (await upgrades.deployProxy(cliptoExchange, [
```

Comment:
Proxy admin here will be able to upgrade contracts / change functionality / drain funds at will. IMO this should not be an EOA, should probably be a multisig of some sort.